app.js

```diff
@@ -30,16 +30,21 @@ const analyticsRouter = require('./app/routes/analytics')
 const wikisRouter = require('./app/routes/wikis')
 const activityRouter = require('./app/routes/activity')
 const ticketRouter = require('./app/routes/ticket')
-
+const passport = require('passport');
 const app = express()
 const server = require('http').Server(app)
+const clientbaseurl = process.env.clientbaseurl ||  'http://localhost:3000'
 
-app.use(cors())
+app.use(cors({origin: clientbaseurl, credentials: true}))
 
 app.use(bodyParser.json({ limit: '200mb' }))
 app.use(cookieParser())
 app.use(bodyParser.urlencoded(fileConstants.fileParameters))
 
+// PassportJS for OAuth
+app.use(passport.initialize());
+require('./app/middleware/passport');
```

Comment:
please make this reusable store this in a variable then use it with the app.use method

Reply:
`passport.initialize()` is already being used in `app.use()`.
All we need to do is pass a middleware in passport to use for requests.
Making is a module will look like this->
```js
//passportOAuth.js
//imports.......
module.exports = {
  initGoogleAuth: () => {
    passport.use(new GoogleStrategy({
      clientID: GOOGLE_OAUTH_CLIENT_ID,
      clientSecret: GOOGLE_OAUTH_CLIENT_SECRET,
      callbackURL: GOOGLE_OAUTH_CALLBACK,
      proxy: true
    }, (accessToken, refreshToken, profile, next) => {

      const provider="google";
      user.findOrCreateForOAuth(profile, provider)
        .then(details => {
          if (details) {
            next(null, details);
          } else {
            next(null, false);
          }
        }).catch(err=>console.log(err))
    }));

  }
}
```
```js
//app.js
const passportOAuth = require('./app/middleware/passportOAuth');
passportOAuth.initGoogleAuth();
```